includes/reader-revenue/class-woocommerce-connection.php

```diff
@@ -24,7 +24,20 @@ class WooCommerce_Connection {
 	 */
 	public static function init() {
 		\add_action( 'admin_init', [ __CLASS__, 'disable_woocommerce_setup' ] );
-		\add_action( 'woocommerce_checkout_order_created', [ __CLASS__, 'register_reader' ] );
+		
+		// WC Subscriptions hooks in and creates subscription at priority 100, so use priority 101.
+		\add_action( 'woocommerce_checkout_order_processed', [ __CLASS__, 'sync_reader_on_order_complete' ], 101 );
+
+		\add_action( 'wp_login', [ __CLASS__, 'sync_reader_on_customer_login' ], 10, 2 );
```

Comment:
This does have some implications for the Simplified Donate Block, since whenever someone logs in it will re-sync the customer data that was synced from Stripe. I think that's probably a good thing though.

Reply:
Not necessarily a good thing :) The `payment_page` meta will be overridden when a "Stripe donor" logs in. I've added a condition so that only non-Newspack-Stripe order will use this sync in 1f5f72943acda5aa68bffe666b64c00e05da5893.